code/datums/beam.dm

```diff
@@ -161,22 +161,31 @@
 	INVOKE_ASYNC(newbeam, TYPE_PROC_REF(/datum/beam, Start))
 	return newbeam
 
-/proc/zap_beam(atom/source, zap_range, damage, list/blacklistmobs)
-	for(var/mob/living/carbon/xenomorph/living AS in cheap_get_xenos_near(source, zap_range))
+/// Shocks xenos, returns shocked xenos.
+/proc/zap_beam(atom/source, zap_range, damage, list/blacklistmobs, max_targets = INFINITY)
+	var/list/xenos = cheap_get_xenos_near(source, zap_range)
+	if(!length(xenos))
+		return list()
```

Comment:
```suggestion
```
cheap get always returns a list so

Reply:
did this because i was having issues with cut